homeassistant/bootstrap.py

```diff
@@ -96,6 +99,25 @@ def _setup_component(hass, domain, config):
                 domain, ", ".join(missing_deps))
             return False
 
+        if hasattr(component, 'CONFIG_SCHEMA'):
+            try:
+                config = component.CONFIG_SCHEMA(config)
+            except vol.MultipleInvalid as ex:
+                _LOGGER.error('Invalid config for [%s]: %s', domain, ex)
+                return False
+
+        if hasattr(component, 'PLATFORM_SCHEMA'):
+            platforms = []
+            for _, platform in config_per_platform(config, domain):
+                try:
+                    platforms.append(component.PLATFORM_SCHEMA(platform))
+                except vol.MultipleInvalid as ex:
+                    _LOGGER.error('Invalid platform config for [%s]: %s. %s',
+                                  domain, ex, platform)
+                    return False
+
+            config = {domain: platforms}
```

Comment:
Good point, line 109 should be an `elif`. You either validate CONFIG_SCHEMA or all platforms against PLATFORM_SCHEMA.

Validating a config will also  transform the config, hence the assignment to the config variable and not having the option to do both.